lib/internal/assert/assertion_error.js

```diff
@@ -41,281 +35,185 @@ const kReadableOperator = {
+  const stringsLen = actual.length + expected.length;
+  if (stringsLen <= kMaxShortStringLength) {
+    return { message: `${actual} !== ${expected}`, header: '' };
   }
 
-  let printedLines = 0;
-  let identical = 0;
-  const msg = `${getErrorMessage(operator, message)}\n${colors.green}+ actual${colors.white} ${colors.red}- expected${colors.white}`;
-  const skippedMsg = ` ${colors.blue}...${colors.white} Lines skipped`;
-
-  let lines = actualLines;
-  let plusMinus = `${colors.green}+${colors.white}`;
-  let maxLength = expectedLines.length;
-  if (actualLines.length < maxLines) {
-    lines = expectedLines;
-    plusMinus = `${colors.red}-${colors.white}`;
-    maxLength = actualLines.length;
+  const isStringComparison = typeof originalActual === 'string' && typeof originalExpected === 'string';
+  // colored myers diff
+  if (isStringComparison && colors.hasColors) {
+    return getColoredMyersDiff(actual, expected);
   }
 
-  for (i = 0; i < maxLines; i++) {
-    if (maxLength < i + 1) {
-      // If more than two former lines are identical, print them. Collapse them
-      // in case more than five lines were identical.
-      if (identical > 2) {
-        if (identical > 3) {
-          if (identical > 4) {
-            if (identical === 5) {
-              res += `\n  ${lines[i - 3]}`;
-              printedLines++;
-            } else {
-              res += `\n${colors.blue}...${colors.white}`;
-              skipped = true;
-            }
-          }
-          res += `\n  ${lines[i - 2]}`;
-          printedLines++;
-        }
-        res += `\n  ${lines[i - 1]}`;
-        printedLines++;
-      }
-      // No identical lines before.
-      identical = 0;
-      // Add the expected line to the cache.
-      if (lines === actualLines) {
-        res += `\n${plusMinus} ${lines[i]}`;
-      } else {
-        other += `\n${plusMinus} ${lines[i]}`;
-      }
-      printedLines++;
-    // Only extra actual lines exist
-    // Lines diverge
+  return getStackedDiff(actual, expected);
+}
+
+function createErrDiff(actual, expected, operator, customMessage) {
+  operator = checkOperator(actual, expected, operator);
+
+  let skipped = false;
+  let message = '';
+  const inspectedActual = StringPrototypeSplit(inspectValue(actual), '\n');
+  const inspectedExpected = StringPrototypeSplit(inspectValue(expected), '\n');
+  const isSimpleDiff = inspectedActual.length === 1 && inspectedExpected.length === 1;
+  let header = `${colors.green}+ actual${colors.white} ${colors.red}- expected${colors.white}`;
+
+  if (isSimpleDiff) {
+    const simpleDiff = getSimpleDiff(actual, inspectedActual[0], expected, inspectedExpected[0]);
+    message = simpleDiff.message;
+    if (typeof simpleDiff.header !== 'undefined') {
+      header = simpleDiff.header;
+    }
+    if (simpleDiff.skipped) {
+      skipped = true;
+    }
+  } else if (areSameObject(inspectedActual, inspectedExpected)) {
```

Comment:
Nit: AFAIK this could be simplified by just comparing inspectedActual / inspectedExpected without splitting the output.